### ISSUE TITLE
Event before Ops flow problem?

### ISSUE THREAD

**Issue by MrSkinny** [sellmerfud/awakening#4]
Excellent work on this script!

I might have been playing this wrong, but I thought as rules written, when a player plays an opponent-associated Event card, they can see the Event resolve BEFORE declaring the Op and Country Target.

This can be critical because you often want the event to resolve before you decide what to do with the Ops points. I know I can manually adjust the game world afterwards, just wanted to see if there was a reason the script required Ops selection first?

**Comment by sellmerfud** [comment]
​When the Bot plays an event that is Associated with the player faction the Event does not trigger.  See Rule 12.4.

When the player plays an event that is Associated with the Bot, after the player chooses which type of action they will take, the program then prompts with:

The card event may trigger, which should happen first?
----------------------------------------------------
1) Operations
2) Event: The door of Itjihad was closed
Selection:

You then enter 1 or 2 depending on what you wish to happen first.

Are you not seeing this?  Are you sure that the particular event you are playing will actually trigger given the current state of the game?

**Comment by MrSkinny** [comment]
My example is say I am US player and enter `us 999` which is an Opponent-Associated event that will place a Plot on board. The program says:

```
You have 3 Ops available and 0 Ops in reserve
Available actions: woi, deploy, [etc...]
```

I will want the Event to trigger, and THEN want to play the **Alert** action.  But I can't select Alert because it is not an available action at this time. This may not be allowed in the board game either, but I've always played that you play the card, tell the opponent to resolve the event, then you play your Operations which whatever Op Type is available **at that moment**.

**Comment by sellmerfud** [comment]
OK, thanks for clarifying.   I see the issue now.

The program currently waits until you select a command before asking if you want the event to trigger first.  This was done because you may want to do something such as log, status, quit, etc. which would not involve the event at all.   

Then once you select a command it prompts for the order.   The works in most cases,
but as you have pointed out, the status of the board may change during the event and can affect your choice of command.

I will have to take deeper look to see how I can rectify this.  Perhaps if the current card it an event associated with the Bot,  the menu of choices will change to include "Perform Jihadist event" and "Do US command" along with log, status, quit, etc.
Then once you select one of those options it will prompt for the specific command you wish to take.

**Comment by MrSkinny** [comment]
Cool, it's interesting because I think this does bring up a valid ambiguity that I certainly hadn't noticed before. The rules say:

```
6.3.1 The player selects a type of Operation and uses the value of the card to implement it

6.3.2 Triggering the Other Side’s Events. Play of a card for Operations that has an Event associated with the opposing side immediately causes the Event to occur.
```

It's arguable that you **do** have to announce the Operation first, then the Event is an interrupt if you choose it to be.  If that's the case, you might be locked into the Op Type, but you are still allowed to declare an "Alert" without a target and if after the Event resolves there's no valid target, you lose the Op points.

I'll ask the groupthink on BGG as well, but I do think regardless of the language, the spirit/intention of the rule is to slightly advantage the current player in being able to react to the event with the most flexibility... since they're burdened by the mandatory event in the first place!

**Comment by sellmerfud** [comment]
Great.  Let me know what you discover.

**Comment by sellmerfud** [comment]
I've been following the bgg thread that you started.  The consensus seems to be that you declare up front whether the event or operations come first.  If you elect to have the event trigger first then the subsequent operation can respond the the effects of the event.

To be honest that is how I have always played the game face to face.  I guess it never came up in my testing so I didn't notice that I had got it wrong in the code.

I have released version 3.2.  Now in the case where the human player plays an event associated with the Bot or plays the US elections, the program first prompts for the order in which the event and operations should take place.

This happens regardless of whether or not the event will actually trigger.  I was trying to be too clever before and not bother prompting if the event would not trigger.   And it turns out that was not a good idea.  Now the user has to make a decision and thus must be paying attention to know whether allowing the event to trigger first is a good idea or not.  To me, that make things more interesting.

Of course when playing as the Jihadist and you have not yet played a plot in the current turn, you are asked if you want to cancel the event (not US elections) by playing a plot operation.

And when playing as the US, it is possible that you are doing a reasssement and (god forbid) you are playing to 3Op cards with Jihadist associated events.  The program handles all of the cases correctly.

Thanks for reporting the problem.  Hopefully I've got it right now.
